### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by Anand8317**
- Person class. It has two methods of_age and can_use_services
- Student class. It is inherited from person class and has one method play_hooky
- Teacher class. It also inherits from person class and overrides method can_use_services of person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    else
+      @parent_permission ? true : false
+    end
+  end
+
+  private
+
+  def of_age?
+    @age > 18
+  end
```
Comment (danmainah) [inline person.rb]: - [ ] @Anand8317 Great work on your person class, however the ` of_Age?` method should return true if the user is 18 years and above in your case its checking if the user is above 18 years.
- [ ] [optional] You can also reduce the size of your `can_use_services` method by simply changing it to `def can_use_services? 
    of_age?() || @parent_permission
  end
`

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Anand8317 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added the person student and teacher classes, each on its file.
✔️  Ensure the teacher and student class are inheriting from the person class.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 1 above)
Comment (Anand8317) [reply person.rb]: hi @danmainah, '@age > 18' returns true or false after checking the condition. So of_age? returns either true or false.
Thank you for the 2nd point.

**Review by Ibizugbe — COMMENTED** [review]
Hi @Anand8317 👋🏼👋🏼👋🏼, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have the right link. This link is meant to be for the previous milestone `OOP school library: add Person, Student and Teacher classes` but you have submitted it for `OOP school library: decorate a class` kindly do well to carefully submit the right link.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age > 18
+  end
```
Comment (Ibizugbe) [inline person.rb]: @Anand8317 , Just as the previous reviewer has suggested, it is a requirement that the off_age? method returns true if the user is 18 years and above (@age >= 18) but your implementation returns true only if a user is above 18 years (@age >18). this means it returns false if the user is 18 years which doesn't align with the project requirements😉, kindly do well to make the changes😉.

image

```suggestion
  def of_age?
    @age >= 18
  end
```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Anand8317 👋🏼👋🏼👋🏼.

require_changes
 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Nice job adhering to the optional change from the previous reviewer👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] Kindly ensure to always write descriptive and professional commit messages as this is a display of professionalism😉. your last commit message didn't describe the feature that was implamented. kindly take note of this for future references👍🏼.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Anand8317) [reply person.rb]: oh my god. Totally embarrassed. I misread it. Thank you.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
```
Comment (FrederickMih) [inline person.rb:26]: :+1:

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Anand8317 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Implementation of `of_age?` method has been rectify :+1: 
`
## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @Anand8317 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Implementation of `of_age?` method has been rectify :+1: 
`
## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Ibizugbe) [reply person.rb]: You don't have to be embarrassed😅 we are all learning and we might encounter such little mistakes as we progress 😅 But let's make better mistakes as we grow together.

Success on your journey.

Happy coding👍.

**Pull request merged**